src/Missings.jl

```diff
@@ -165,4 +166,46 @@ function levels(x)
     levs
 end
 
+struct Conditional{P,X,Y} <: Function end
+
+"""
+    conditional(predicate, x, y)
+
+Return a function that that applies function `predicate` to its positional and keyword
+arguments and returns the value of `x` applied to those arguments if `predicate`
+returns true and otherwise returns the value of `y` applied to those arguments.
+
+# Examples
+```jldoctest
+julia> f = conditional(x -> x ≥ 0, sqrt, x -> sqrt(complex(x)));
+
+julia> f.([4, -4])
+2-element Array{Number,1}:
+     2.0
+      0.0 + 2.0im
+"""
+conditional(predicate::Function, x::Base.Callable, y::Base.Callable) =
+    Conditional{predicate, x, y}()
+(::Conditional{P,X,Y})(xs...;kw...) where {P,X,Y} =
+    P(xs...; kw...) ? X(xs...; kw...) : Y(xs...; kw...)
+
+_passmissing_predicate(xs...;kw...) =
+    any(ismissing.(xs)) || any(ismissing.(values(values(kw))))
```

Comment:
It's more efficient to pass `ismissing` as a function argument to `any` rather than broadcasting, i.e.
```julia
    any(ismissing, xs) || any(ismissing, values(values(kw)))
```

Reply:
Right. In any case, this really needs to be unrolled, specializing on the number of arguments, or it will be too slow.